persistence/jpa/src/main/java/org/infinispan/persistence/jpa/configuration/JpaStoreConfigurationBuilder.java

```diff
@@ -50,6 +54,10 @@ public JpaStoreConfigurationBuilder storeMetadata(boolean storeMetadata) {
 
    @Override
    public void validate() {
+      Boolean segmented = attributes.attribute(SEGMENTED).get();
+      if (segmented == null || segmented) {
+         throw CONFIG.storeDoesNotSupportBeingSegmented(JpaStore.class.getSimpleName());
+      }
```

Comment:
Shouldn't we set SEGMENTED to false automatically in the builder of store's which don't support segmented? That way users don't have to explicitly set it to false.

Reply:
You actually suggested this behavior :) 

> Ryan Emerson: I'm in favour of being explicit and throwing the error
> Ryan Emerson: That might make the user more aware of the limitations of a non-segmented store

Reply:
O yeah :see_no_evil:  ... sorry about that. Lets stick with what you have then :slightly_smiling_face: